Source/UI/CommonUI/Protocols/Elements/ElementBorderStyle.swift

```diff
@@ -0,0 +1,11 @@
+import UIKit
+
+public protocol ElementBorderStyle {
+    var cornerRadius: CGFloat { get }
+    var borderWidth: CGFloat { get }
+    var normalColor: UIColor { get }
+    var focusColor: UIColor { get }
+    var errorColor: UIColor { get }
+    var edges: UIRectEdge? { get }
+    var corners: UIRectCorner? { get }
```

Comment:
The problem with UIRectCorner is it does not currently support `automatic`.

We have different options for setting up the corners:
- selected corners (topLeft, topRight, bottomLeft, bottomRight). This is supported by UIRectCorner
- allCorners. Also supported by UIRectCorner
- autodetect / automatic / auto. This would be matched by the edges requested and imo should be the default on all our implementations. For example if edges is top, right & bottom, this corner mode will add corners to topRight & bottomRight. I don't think this can be achieved as intended without the new option.

We can however create 
```swift
public extension UIRectCorner {
    static let automatic = Self(rawValue: 1 << 4)
}
```

The problem with doing so is that any Frames consumer will see this changed native interface and may wrongly assume this is native. Which is why personally I would have favoured us creating our own optionSet for this, showing a clear ownership over the change and not interfering with Apple library.